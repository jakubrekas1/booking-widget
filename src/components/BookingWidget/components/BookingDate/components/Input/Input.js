import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ onClick, placeholder, value }) => (
  <input disabled onClick={onClick} placeholder={placeholder} value={value} />
);

Input.defaultProps = {
  placeholder: '',
  value: '',
};

Input.propTypes = {
  onClick: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
};

export default Input;
