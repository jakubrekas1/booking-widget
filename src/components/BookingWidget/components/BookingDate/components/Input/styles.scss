@import '../../../../../../assets/styles/colors';

.booking-date{
  &__field{
    width: 50%;
    padding: 4px 8px;
    font-size: 22px;
    color: $black;
    background-color: $white;
    border-width: 0;
    border-radius: 4px;
    cursor: pointer;
    transition: 100ms all linear;

    &:hover,
    &.active{
      background-color: $base--light;
      color: $base--dark;
    }
  }
}
