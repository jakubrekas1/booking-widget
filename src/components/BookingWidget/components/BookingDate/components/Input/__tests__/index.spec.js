import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

describe('Input component', () => {
  it('should renders correctly', () => {
    const Component = <Input onClick={() => null} placeholder="Some placeholder..." value="2019/12/15" />;
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders correctly with default props', () => {
    const Component = <Input onClick={() => null} />;
    const tree = renderer.create(Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
