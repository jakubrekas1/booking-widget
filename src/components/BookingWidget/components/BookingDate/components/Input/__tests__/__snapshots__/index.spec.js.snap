// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should renders correctly 1`] = `
<input
  disabled={true}
  onClick={[Function]}
  placeholder="Some placeholder..."
  value="2019/12/15"
/>
`;

exports[`Input component should renders correctly with default props 1`] = `
<input
  disabled={true}
  onClick={[Function]}
  placeholder=""
  value=""
/>
`;
